src/components/organisms/UiAccordion/_internal/UiAccordionItem.vue

```diff
@@ -71,98 +71,97 @@ import {
 import UiListItem from '../../UiList/_internal/UiListItem.vue';
+import type { ListItemAttrsProps } from '../../UiList/_internal/UiListItem.vue';
 import type {
-  AccordionValue,
-  AccordionItemSettings,
+  AccordionModelValue,
+  AccordionToggle,
 } from '../UiAccordion.vue';
-import type { IconName } from '../../../../types/icon';
+import type {
+  DefineAttrsProps,
+  Icon,
+} from '../../../../types';
 
-const props = defineProps({
+export interface AccordionItemSettings {
+  iconOpen?: Icon;
+  iconClose?: Icon;
+}
+export interface AccordionItemProps {
   /**
     * Use this props to set item title.
     */
-  title: {
-    type: String,
-    default: '',
-  },
+  title?: string;
   /**
     * Use this props to set item name, it used to toggle.
     */
-  name: {
-    type: String,
-    default: '',
-  },
+  name?: string;
   /**
     * Use this props to setup item component.
     */
-  settings: {
-    type: Object as PropType<AccordionItemSettings>,
-    default: () => ({
-      iconOpen: 'chevron-up',
-      iconClose: 'chevron-down',
-    }),
-  },
+  settings?: AccordionItemSettings;
   /**
    *  Use this props to pass attrs to toggler UiButton.
    */
-  buttonTogglerAttrs: {
-    type: Object,
-    default: () => ({}),
-  },
+  buttonTogglerAttrs?: ButtonAttrsProps;
   /**
    *  Use this props to pass attrs to toggler UiIcon.
    */
-  iconTogglerAttrs: {
-    type: Object,
-    default: () => ({}),
-  },
+  iconTogglerAttrs?: IconAttrsProps;
   /**
-   *  Use this props to pass attrs to content element.
+   *  Use this props to pass attrs to list item.
    */
-  contentAttrs: {
-    type: Object,
-    default: () => ({}),
-  },
+  listItemAttrs?: ListItemAttrsProps['listItemAttrs'];
   /**
-   *  Use this props to pass attrs to list item.
+   *  Use this props to pass attrs to content element.
    */
-  listItemAttrs: {
-    type: Object,
-    default: () => ({}),
-  },
+  contentAttrs?: DefineAttrsProps<null>;
+}
+export type AccordionItemAttrsProps = DefineAttrsProps<AccordionItemProps, ListItemAttrsProps>;
+
+const props = withDefaults(defineProps<AccordionItemProps>(), {
+  title: '',
+  name: '',
+  settings: () => ({
+    iconOpen: 'chevron-up',
+    iconClose: 'chevron-down',
+  }),
+  buttonTogglerAttrs: () => ({}),
+  iconTogglerAttrs: () => ({ icon: 'chevron-down' }),
+  contentAttrs: () => ({}),
+  listItemAttrs: () => ({ class: 'ui-accordion-item' }),
 });
-const opened = inject('opened') as ComputedRef<AccordionValue>;
-const toggle = inject('toggle') as (name: string) => void;
-const isOpen = computed(() => {
-  if (opened.value === 'string') {
-    return props.name === opened.value;
-  }
-  return opened.value.includes(props.name);
+const opened = inject<ComputedRef<AccordionModelValue>>('opened', computed(() => ''));
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const toggle = inject<AccordionToggle>('toggle', () => {});
```

Comment:
Q: Why is this function empty? :D And Why does it return either empty promise or nothing? Can it be written as `() => undefined` instead of `() => {}`? Would that solve an eslint problem? :D